common/sealing/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+//! A crate that facilitates sealing using ChaCha20-Poly1305, an algorithm for
+//! [AEAD][aead].
+//!
+//! [aead]: https://en.wikipedia.org/wiki/Authenticated_encryption
+//! [chacha]: https://en.wikipedia.org/wiki/ChaCha20-Poly1305
+#![no_std]
+
+extern crate alloc;
+
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+
+use ring_compat::aead::{AeadInPlace, ChaCha20Poly1305, KeyInit};
+use ring_compat::generic_array::typenum::{U12, U32};
+use ring_compat::generic_array::GenericArray;
+pub use zeroize::Zeroizing;
+use zeroize::{Zeroize, ZeroizeOnDrop};
+
+/// A nonce should only ever be used once for any given key, across your entire
+/// code base.  Nonces may either be randomly generated or, otherwise, carefully
+/// selected to ensure uniqueness.
+#[derive(Debug)] // core
+#[derive(Zeroize, ZeroizeOnDrop)] // zeroize
+pub struct Nonce([u8; 12]);
+
+impl Nonce {
+    pub const SIZE: usize = 12;
+
+    pub fn new(nonce: [u8; Self::SIZE]) -> Self {
+        Nonce::from(nonce)
+    }
+}
+
+#[derive(Debug, Clone)] // core
+#[derive(Zeroize, ZeroizeOnDrop)] // zeroize
+/// A key used for symmetric encryption.
+pub struct SecretKey([u8; 32]);
+
+impl SecretKey {
+    pub const SIZE: usize = 32;
+
+    pub fn new(key: [u8; Self::SIZE]) -> Self {
+        SecretKey::from(key)
+    }
+}
+
+impl From<[u8; SecretKey::SIZE]> for SecretKey {
+    fn from(secret: [u8; SecretKey::SIZE]) -> Self {
+        SecretKey(secret)
+    }
+}
+
+impl From<[u8; Nonce::SIZE]> for Nonce {
+    fn from(nonce: [u8; Nonce::SIZE]) -> Self {
+        Nonce(nonce)
+    }
+}
+
+impl From<SecretKey> for [u8; SecretKey::SIZE] {
+    fn from(SecretKey(bytes): SecretKey) -> Self {
+        bytes
+    }
+}
+
+impl From<Nonce> for [u8; Nonce::SIZE] {
+    fn from(Nonce(bytes): Nonce) -> Self {
+        bytes
+    }
+}
+
+type NonceSize = U12;
+type SecretKeySize = U32;
+
+/// Cryptographically seal and authenticate a message using the
+/// [ChaCha20-Poly1305][chacha20poly1305] algorithm for [AEAD][aead].
+/// Associated data is also authenticated, but not encrypted.
+///
+/// [aead]: https://en.wikipedia.org/wiki/Authenticated_encryption
+/// [chacha]: https://en.wikipedia.org/wiki/ChaCha20-Poly1305
```

Comment:
```suggestion
/// [ChaCha20-Poly1305] algorithm for [AEAD].
/// Associated data is also authenticated, but not encrypted.
///
/// [AEAD]: https://en.wikipedia.org/wiki/Authenticated_encryption
/// [ChaCha20-Poly1305]: https://en.wikipedia.org/wiki/ChaCha20-Poly1305
```
- reference was broken
- this markup allows such shortcuts

Reply:
you probably missed to do this change

Reply:
I believe I accidentally undid the change during a rebase.  Should be rectified now though.